File: codility/ BinaryGap.js
// Find longest sequence of zeros in binary representation of an integer.

function solution(N) {
    // write your code in JavaScript (Node.js 8.9.4)
    const bin = N.toString(2);
    const arr = bin.split('1');
    let maxGap = 0;
    for (let i = 1; i < arr.length - 1; i++) {
        maxGap = Math.max(maxGap, arr[i].length);
    }
    return maxGap;
}

File: codility/binary_gap.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(Solution(1041))
}

func Solution(N int) int {
	count := 0
	result := 0
	trailingZeros := true

	for i := N; i != 0; i >>= 1 {
		if i&1 == 1 {
			if trailingZeros {
				trailingZeros = false
			} else {
				result = max(count, result)
				count = 0
			}

		} else {
			count++
		}
	}
	return result
}

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}
